src/libs/auth.js

```diff
@@ -21,26 +21,29 @@ import * as Utils from 'src/libs/utils'
 
 
 export const getOidcConfig = () => {
+  const metadata = {
+    authorization_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
+    token_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/token`,
+    ...(isGoogleAuthority() && { userinfo_endpoint: 'https://openidconnect.googleapis.com/v1/userinfo' })
+  }
   return {
     authority: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
     client_id: authStore.get().oidcConfig.clientId,
     popup_redirect_uri: `${window.origin}/redirect-from-oauth`,
     silent_redirect_uri: `${window.origin}/redirect-from-oauth-silent`,
-    prompt: 'login',
+    metadata,
+    prompt: 'consent',
     scope: 'openid email profile',
-    metadata: {
-      authorization_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
-      token_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/token`
-    },
+    loadUserInfo: isGoogleAuthority(),
     userStore: new WebStorageStateStore({ store: window.localStorage }),
-    accessTokenExpiringNotificationTimeInSeconds: 300,
```

Comment:
@rtitle can you say more about the removal of `accessTokenExpiringNotificationTimeInSeconds`? I'm also having trouble understanding where the refresh token is requested.

Reply:
Yes. We actually use 2 callbacks:
1. `accessTokenExpiring` -- this fires shortly _before_ the access token expires. In this case we silently renew the access token without needing the refresh token. This is not always reliable (e.g. it can be missed if the browser is asleep).
2. `accessTokenExpired` -- this fires _after_ the access token expires. In this case the refresh_token is used to get a new access_token.

I could hear an argument that (1) isn't necessary if we're just using a refresh token anyway.

In any case this parameter only affects case (1). I had previously increased the time-before-expire to 5 minutes in the hope that it would be more reliable. Now it's using the default of 2 minutes I believe.

Reply:
Also the refresh token is requested via:
```
extraQueryParams: { access_type: 'offline' }
```

Reply:
Thanks, but I don't actually see `accessTokenExpiring` anyplace in the code. Can you point me to where it is?

Reply:
It comes from `automaticSilentRenew: true`: https://github.com/DataBiosphere/terra-ui/pull/3117/files#diff-a4ba38b20453374fed28d8c26c28f5b20b486db965bbfd9c47032a9cb99c122cR39. The library sets the accessTokenExpiring callback [[source code](https://github.com/authts/oidc-client-ts/blob/main/src/SilentRenewService.ts#L23)].

Previously I experimented with setting my own `accessTokenExpiring` callback vs letting the library do it, but the library version seems to work fairly well (it has some retries in case of errors, etc).

Reply:
Got it, thank you!